### DISCUSSION TITLE
Requirements for hallmarked nodes

### DISCUSSION THREAD

**Discussion by florintimbuc** [CoopNetwork/GMD-blockchain#12] [Q&A] [answered]
Hi guys,
I would have some questions before launching a few nodes for COOP Network. I didn’t find these aspects written in README or on your website.

1. What do you recommend in terms of system requirements for multiple hallmarked nodes?
	1.1. Dedicated/physical hardware is needed or some virtual servers will simply do the job?
	1.2. Give us some numbers please: CPU cores (Intel or AMD), RAM, Storage (would SSD help?), Ethernet speed & traffic/month plan?

2. Do you have any slashing system implemented to discourage any validator misbehavior? What’s the minimum availability needed for the node to run to maintain it’s eligibility?

**Comment by georgeasiminei** [comment]
Hi @florintimbuc, welcome in our community!

1. The node is known to run on low resources. Dedicated hardware is not required. I have nodes running without problems on VPS with 1 core, 512MB of RAM (although 1GB is prefferred), and 10GB of disk. 
No SSD is required as this is not a disk intensive app. 
Doesn't matter if CPU is Intel or AMD or ARM, as long as it can run Java 11 or newer JVM (basically any mainstream CPU made in the last 10 years works just fine). 
Considering usual traffic a single node can run with 2mbps internet connection. Right now the monthly traffic is in the range of hundreds of MB, but depending on network load, expect to increase. I estimate that no more than 600GB of data per node per month will be required even in highest network loads.

2. There is a built-in automatic system for blacklisting nodes that have a different version of the blockchain. The blacklist can also be imposed manually in the default config file. 
There is no requirement for availability. You can run a node just when you needed for a transaction, for example.

I hope this answers your questions, please let me know if I can clarify further any point above.

**Comment by florintimbuc** [comment]
Got it! Thanks for your in-depth reply. :)
I would like to create a Docker swarm for monitoring & run multiple validators on GMD. Regarding the fact that low resources are needed (1 core, 512MB-1GB), how many nodes would you recommend running in Docker containers on the same 2vCPU & 2GB RAM instance? What about running multiple nodes on a 2vCPU & 4GB RAM instance? Tell me please if is there any perfect ratio CPU:RAM so I can optimise my infrastructure. :)

**Comment by georgeasiminei** [comment, marked as answer]
In your case, RAM would be the bottleneck. 
512MB of RAM work ok only if you have enough swap disk space. With 512 MB of RAM I usually use Debian 10 and add 2GB of swap, to be on safe side.   

Debian 10 has RAM footprint of about ~70-80MB, that would leave the node with only ~420MB which works fine in low load. If you use docker, that means on top of that you add the RAM footprint of docker itself and that of the containers.

In docker, I would run 1 node with 2GB of RAM, and 2 or maximum 3 nodes with 4GB of RAM. CPU is not an issue, 2 cores are fine.
